lib/core/context.c

```diff
@@ -190,3 +190,69 @@ void h2o_context_update_timestamp_string_cache(h2o_context_t *ctx)
     h2o_time2str_rfc1123(ctx->_timestamp_cache.value->rfc1123, &gmt);
     h2o_time2str_log(ctx->_timestamp_cache.value->log, ctx->_timestamp_cache.tv_at.tv_sec);
 }
+
+void h2o_context_close_idle_connections(h2o_context_t *ctx, size_t max_connections_to_close, uint64_t min_age)
+{
+    if (max_connections_to_close <= 0)
+        return;
+
+    size_t closed = ctx->_conns.num_conns.shutdown;
+
+    if (closed >= max_connections_to_close)
+        return;
+
+    H2O_CONN_LIST_FOREACH(h2o_conn_t * conn, ({&ctx->_conns.idle}), {
+        struct timeval now = h2o_gettimeofday(ctx->loop);
+        if (h2o_timeval_subtract(&conn->connected_at, &now) < (min_age * 1000))
+            continue;
```

Comment:
I wonder if we need this logic at all, based on my understanding that we agreed that connections (specifically H1) are not considered as idle when they have been created but have not received any requests.

Though we might want to make sure that view is reflected to the logic that we have in http2 and http3 as well. For the purpose, IIRC h2 and h3 has the concept of largest stream ID being received, or the next stream ID to be received. We can probably use that to determine the state of the connection.

Reply:
H1 are considered as `idle` on creation, `active` once they receive data, then (before parsing a request) `idle` again if there's no more data to process.

I'm not sure I follow the stream id method of determining the state of the connection, could you expand on this idea?

Am I on the right track in thinking:
- If received stream_id > max stream id: idle
- if received stream_id < max stream id: active